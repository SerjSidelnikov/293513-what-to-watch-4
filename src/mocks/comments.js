export default [
  {
    id: `0`,
    author: `Kate Muir`,
    date: `Month dd, yyyy`,
    description: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    rate: `8,9`,
  },
  {
    id: `1`,
    author: `Bill Goodykoontz`,
    date: `Month dd, yyyy`,
    description: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    rate: `7,3`,
  },
  {
    id: `2`,
    author: `Amanda Greever`,
    date: `Month dd, yyyy`,
    description: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    rate: `7,4`,
  },
  {
    id: `3`,
    author: `Matthew Lickona`,
    date: `Month dd, yyyy`,
    description: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    rate: `7,1`,
  },
  {
    id: `4`,
    author: `Paula Fleri-Soler`,
    date: `Month dd, yyyy`,
    description: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    rate: `7,2`,
  },
  {
    id: `5`,
    author: `Paula Fleri-Soler`,
    date: `Month dd, yyyy`,
    description: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    rate: `7,9`,
  },
];
