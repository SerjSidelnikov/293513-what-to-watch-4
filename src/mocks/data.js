export default {
  film: {
    title: `The Grand Budapest Hotel`,
    genre: `Drama`,
    releaseDate: 2014,
  },
  movies: [
    `Fantastic Beasts`,
    `Bohemian Rhapsody`,
    `Macbeth`,
  ],
};
