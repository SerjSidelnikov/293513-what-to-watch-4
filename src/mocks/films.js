export default [
  {
    id: 1,
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    thumbnail: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `genre`,
    releaseDate: `2014`,
    promo: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/a/ab/Caminandes_3_-_Llamigos_-_Blender_Animated_Short.webm/Caminandes_3_-_Llamigos_-_Blender_Animated_Short.webm.360p.vp9.webm`,
  },
  {
    id: 2,
    name: `Bohemian Rhapsody`,
    thumbnail: `img/bohemian-rhapsody.jpg`,
    genre: `genre`,
    releaseDate: `2014`,
    promo: `img/bohemian-rhapsody.jpg`,
    poster: `img/bohemian-rhapsody.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 3,
    name: `Macbeth`,
    thumbnail: `img/macbeth.jpg`,
    genre: `genre`,
    releaseDate: `2014`,
    promo: `img/macbeth.jpg`,
    poster: `img/macbeth.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/a/ab/Caminandes_3_-_Llamigos_-_Blender_Animated_Short.webm/Caminandes_3_-_Llamigos_-_Blender_Animated_Short.webm.480p.vp9.webm`,
  },
  {
    id: 4,
    name: `Aviator`,
    thumbnail: `img/aviator.jpg`,
    genre: `genre`,
    releaseDate: `2014`,
    promo: `img/aviator.jpg`,
    poster: `img/aviator.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 5,
    name: `We need to talk about Kevin`,
    thumbnail: `img/we-need-to-talk-about-kevin.jpg`,
    genre: `genre`,
    releaseDate: `2014`,
    promo: `img/we-need-to-talk-about-kevin.jpg`,
    poster: `img/we-need-to-talk-about-kevin.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/a/ab/Caminandes_3_-_Llamigos_-_Blender_Animated_Short.webm/Caminandes_3_-_Llamigos_-_Blender_Animated_Short.webm.480p.vp9.webm`,
  },
  {
    id: 6,
    name: `What We Do in the Shadows`,
    thumbnail: `img/what-we-do-in-the-shadows.jpg`,
    genre: `genre`,
    releaseDate: `2014`,
    promo: `img/what-we-do-in-the-shadows.jpg`,
    poster: `img/what-we-do-in-the-shadows.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 7,
    name: `Revenant`,
    thumbnail: `img/revenant.jpg`,
    genre: `genre`,
    releaseDate: `2014`,
    promo: `img/revenant.jpg`,
    poster: `img/revenant.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/a/ab/Caminandes_3_-_Llamigos_-_Blender_Animated_Short.webm/Caminandes_3_-_Llamigos_-_Blender_Animated_Short.webm.480p.vp9.webm`,
  },
  {
    id: 8,
    name: `Johnny English`,
    thumbnail: `img/johnny-english.jpg`,
    genre: `genre`,
    releaseDate: `2014`,
    promo: `img/johnny-english.jpg`,
    poster: `img/johnny-english.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
];
