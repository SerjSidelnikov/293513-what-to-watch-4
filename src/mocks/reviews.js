export default [
  {
    id: `0`,
    author: `Kate Muir`,
    date: `December 24, 2016`,
    description: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    rating: `8,9`,
  },
  {
    id: `1`,
    author: `Bill Goodykoontz`,
    date: `December 24, 2016`,
    description: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    rating: `7,3`,
  },
  {
    id: `2`,
    author: `Amanda Greever`,
    date: `December 24, 2016`,
    description: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    rating: `7,4`,
  },
  {
    id: `3`,
    author: `Matthew Lickona`,
    date: `December 24, 2016`,
    description: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    rating: `7,1`,
  },
  {
    id: `4`,
    author: `Paula Fleri-Soler`,
    date: `December 24, 2016`,
    description: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    rating: `7,2`,
  },
  {
    id: `5`,
    author: `Paula Fleri-Soler`,
    date: `December 24, 2016`,
    description: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    rating: `7,9`,
  },
];
