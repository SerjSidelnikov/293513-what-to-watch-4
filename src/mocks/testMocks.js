export const promoItem = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  releaseDate: 2014,
};

export const film = {
  id: 0,
  title: `Test title`,
  poster: `https://picsum.photos/id/0`,
};

export const films = [
  {
    id: 1,
    title: `Test title 1`,
    poster: `https://picsum.photos/id/1`,
  },
  {
    id: 2,
    title: `Test title 2`,
    poster: `https://picsum.photos/id/2`,
  },
];

