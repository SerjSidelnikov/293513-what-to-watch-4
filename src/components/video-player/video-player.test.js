import React from 'react';
import renderer from 'react-test-renderer';

import VideoPlayer from './video-player';
import {film} from '../../mocks/testMocks';

describe(`VideoPlayer`, () => {
  it(`VideoPlayer rendered correctly`, () => {
    const {preview, poster} = film;

    const tree = renderer.create(
        <VideoPlayer
          src={preview}
          poster={poster}
          isPlaying={false}
          muted={true}
        />,
        {
          createNodeMock: () => {
            return {};
          }
        }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
