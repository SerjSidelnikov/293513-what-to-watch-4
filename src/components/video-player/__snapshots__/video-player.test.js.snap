// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayer VideoPlayer rendered correctly 1`] = `
<video
  autoPlay={false}
  height="100%"
  loop={true}
  preload="none"
  width="100%"
/>
`;
